Framework/MDAlgorithms/src/SmoothMD.cpp

```diff
@@ -126,8 +126,8 @@ KernelVector gaussianKernel(const double fwhm) {
 KernelVector renormaliseKernel(KernelVector kernel,
                                std::vector<bool> validity) {
 
-  if (std::accumulate(validity.cbegin(), validity.cend(), 0.0) <
-      kernel.size()) {
+  if (std::accumulate(validity.cbegin(), validity.cend(),
+                      static_cast<size_t>(0)) < kernel.size()) {
```

Comment:
Can this be replaced by checking that `kernel.size() == validity.size()` and `std::find(validity.cbegin(), validity.cend(), [](decltype(kernel)::reference b){ return b == false;}) != validity.cend()`?